delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -213,16 +223,25 @@ class ElasticSearchClient(client: HttpClient, endpoint: Uri)(implicit as: ActorS
   )(
       sort: SortList = SortList.empty
   ): HttpResult[Json] = {
-    val searchEndpoint = (endpoint / indexPath(indices) / searchPath).withQuery(Uri.Query(defaultQuery ++ qp.toMap))
-    val payload        = QueryBuilder(query).withSort(sort).withTotalHits(true).build
-    client.toJson(Post(searchEndpoint, payload))
+    if (indices.isEmpty)
+      IO.pure(emptyResults)
+    else {
+      val (indexPath, q) = indexPathAndQuery(indices, QueryBuilder(query))
+      val searchEndpoint = (endpoint / indexPath / searchPath).withQuery(Uri.Query(defaultQuery ++ qp.toMap))
+      val payload        = q.withSort(sort).withTotalHits(true).build
+      client.toJson(Post(searchEndpoint, payload))
+    }
   }
 
   private def discardEntity(resp: HttpResponse) =
     UIO.delay(resp.discardEntityBytes()) >> IO.unit
 
-  private def indexPath(indices: Set[String]): String =
-    if (indices.isEmpty) allIndexPath else indices.mkString(",")
+  private def indexPathAndQuery(indices: Set[String], query: QueryBuilder): (String, QueryBuilder) = {
+    indices.toList match {
+      case single :: Nil => (single, query)
+      case more          => (allIndexPath, query.withIndices(more))
```

Comment:
what about the:
```scala
case Nil => ???
```

Reply:
it shouldn't happen, since we check for empty before calling this function. If it's empty we'll call `_all` with empty `_index` `terms` filter, which will mean no indices will match